File: DSA-DAILY/30-07-2025/2419. Longest Subarray With Maximum Bitwise AND Leetcode-med.java
class Solution {
    public int longestSubarray(int[] nums) {
        int ans=0,s=0,maxs=0;
        for(int num:nums){
            if(num>maxs){
                maxs=num;
                s=ans=0;
            }
            if(maxs==num){
                s++;
            }else{
                s=0;
            }
            if(ans<s)ans=s;
        }   
        return ans;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {1, 2, 3, 4, 3, 2, 1};
        int result = solution.longestSubarray(nums);
        System.out.println(result); // Output: 4
    }
}

File: DSA-DAILY/30-07-2025/2419. Longest Subarray With Maximum Bitwise AND Leetcode-med.py
from typing import List
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maxs=ans=s=0
        for num in nums:
            if num > maxs:
                ans=s=0
                maxs=num
            if num == maxs:
                s+=1
            else:
                s=0
            ans=max(ans,s)
        return ans
    
if __name__ == "__main__":
    sol = Solution()
    nums = [1, 2, 4, 4, 4, 2, 1]
    print(sol.longestSubarray(nums))
